File: Inverted-Index/src/Exclude.java
import java.util.List;
import java.util.Map;
import java.util.Set;


class Exclude extends Operation {

    Exclude(List<String> words, Map<String, List<Doc>> map){
        super(words, map);
    }

    @Override
    public Set<Doc> operate(Set<Doc> docs) {
        for (String word : keyWords) {
            var list = map.get(word);
            if(list != null) {
                docs.removeAll(list);
            }
        }
        return docs;
    }
}

File: Phase5-TDD/InvertedIndex/Keywords/Exclude.cs
using System;
using System.Collections.Generic;

namespace InvertedIndex.Keywords
{
    public class Exclude : Keywords
    {
        private string word;

        public Exclude(string word)
        {
            this.word = word;
        }

        public string getWord()
        {
            return word;
        }
        public override HashSet<Doc> operate(HashSet<Doc> docs, HashSet<Doc> newDocs)
        {
            throw new NotImplementedException();
        }
    }
}